server/src/main/java/com/android/identity/wallet/server/VerifierServlet.kt

```diff
@@ -967,43 +971,12 @@ lrW+vvdmRHBgS+ss56uWyYor6W7ah9ygBwYFK4EEACI=
                 kvPairs[parts[0]] = parts[1]
             }
 
-            val response = kvPairs["response"]
-            val encryptedJWT = EncryptedJWT.parse(response)
-
-            val encPub = session.encryptionKey.publicKey.javaPublicKey as ECPublicKey
-            val encPriv = session.encryptionKey.javaPrivateKey as ECPrivateKey
-            val encKey = ECKey(
-                Curve.P_256,
-                encPub,
-                encPriv,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null,
-                null
-            )
-
-            val decrypter = ECDHDecrypter(encKey)
-            encryptedJWT.decrypt(decrypter)
-
-            val vpToken = encryptedJWT.jwtClaimsSet.getClaim("vp_token") as String
-            if (session.requestFormat == "mdoc") {
-                session.deviceResponse = vpToken.fromBase64Url()
-            } else {
-                session.deviceResponse = vpToken.toByteArray()
+            if (session.responseMode == "direct_post.jwt") {
+                updateSessionFromDirectPostJwtResponse(kvPairs, session)
+            } else if (session.responseMode == "direct_post") {
```

Comment:
Same suggestion here. In addition, there seem to be also the `else` case not covered now (neither of two). Is there such a possibility?